spring/boot-tomcat-sample/src/main/java/com/linecorp/armeria/spring/tomcat/demo/SpringTomcatApplication.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.spring.tomcat.demo;
+
+import javax.inject.Inject;
+
+import org.apache.catalina.startup.Tomcat;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.web.context.WebServerApplicationContext;
+import org.springframework.boot.web.embedded.tomcat.TomcatWebServer;
+import org.springframework.boot.web.server.WebServer;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+
+import com.linecorp.armeria.server.tomcat.TomcatService;
+import com.linecorp.armeria.spring.ArmeriaServerConfigurator;
+
+@SpringBootApplication
+public class SpringTomcatApplication {
+
+    @Inject
+    private ApplicationContext applicationContext;
+
+    /**
+     * Bean to configure Armeria Tomcat service.
+     * @return configuration bean.
+     */
+    @Bean
+    public ArmeriaServerConfigurator armeriaTomcat() {
+        WebServer webServer = ((WebServerApplicationContext) applicationContext).getWebServer();
+        if (webServer instanceof TomcatWebServer) {
+            Tomcat tomcat = ((TomcatWebServer) webServer).getTomcat();
+
+            return serverBuilder -> serverBuilder.service("prefix:/tomcat/api/rest/v1",
+                                                          TomcatService.forTomcat(tomcat));
+        }
+        return serverBuilder -> { };
+    }
+
+    /**
+     * Main method.
+     * @param args program args.
+     */
+    @SuppressWarnings("checkstyle:uncommentedmain")
```

Comment:
Could we fix this violation rather than suppressing it?

Reply:
Also wonder for this type of reason and just avoiding building the example every time we build armeria, should we create a separate `armeria-examples` repo?

Reply:
Yes the main method is needed to launch spring boot. Do you mean adding to `checkstyle-suppressions.xml`

I think a separate repo is a good idea, except that since an example project like this is meant to reproduce issues, if Armeria breaks an example, then the Armeria build should fail also IMO.

Reply:
Isn't an example to just show how to use the code? If we have integration test cases we could add to the `it` module

Reply:
I was thinking the example could function as both. But I also see your point.

since `it` is a single project, I think we need to refactoring `it` to have several sub-projects